util/resource/summary.go

```diff
@@ -28,9 +28,14 @@ func (ss Summaries) Duration() wfv1.ResourcesDuration {
 	// Add container states.
 	d := wfv1.ResourcesDuration{}
 	for _, s := range ss {
+		// age is converted to seconds, otherwise the multiplication below is very likely to overflow
 		age := int64(s.age().Seconds())
 		for n, q := range s.ResourceList {
-			d = d.Add(wfv1.ResourcesDuration{n: wfv1.NewResourceDuration(time.Duration(q.Value() * age / wfv1.ResourceQuantityDenominator(n).Value() * int64(time.Second)))})
+			d = d.Add(wfv1.ResourcesDuration{
+				n: wfv1.NewResourceDuration(time.Duration(
+					q.MilliValue()*age/wfv1.ResourceQuantityDenominator(n).MilliValue(),
+				) * time.Second),
```

Comment:
so is the previous `int64` conversion on `time.Second` here no longer necessary?

Reply:
It is still necessary, I've just moved it out to the next line to avoid the cast to `int64`, and to make sure that the division by the `ResourceQuantityDenominator.MilliValue()` (which is `1e11` for memory) is done before the multiplication by `time.Second` to ensure we don't overflow.

Reply:
Sorry, looking back at this I think I read your question too fast previously. 
The cast to `int64` is no longer necessary, since everything on line 36 is casted to a `time.Duration`, we can multiply directly with `time.Second` which is also of type `time.Duration`.

Reply:
Gotcha thanks for the clarification there, just wanted to make sure.

Rephrasing, before it was multiplied inside of the `time.Duration(...)` and now it's outside of that to avoid the overflow as you mentioned. Since it's outside now, it no longer needs to cast to an `int64`